includes/classes/Indexable/Post/SyncManager.php

```diff
@@ -94,6 +96,134 @@ public function action_queue_meta_sync( $meta_id, $object_id, $meta_key, $meta_v
 		}
 	}
 
+	/**
+	 * When a term is updated, re-index all posts attached to that term
+	 *
+	 * @param  int       $term_id Term id.
+	 * @param  int       $tt_id Term Taxonomy id.
+	 * @param  string    $taxonomy Taxonomy name.
+	 * @since  3.5
+	 */
+	public function action_edited_term( $term_id, $tt_id, $taxonomy ) {
+		global $wpdb;
+
+		if ( defined( 'DOING_AUTOSAVE' ) && DOING_AUTOSAVE ) {
+			// Bypass saving if doing autosave
+			return;
+		}
+
+		// Find ID of all attached posts (query lifted from wp_delete_term())
+		$object_ids = (array) $wpdb->get_col( $wpdb->prepare( "SELECT object_id FROM $wpdb->term_relationships WHERE term_taxonomy_id = %d", $tt_id ) );
```

Comment:
Instead of direct SQL, can't posts be retrieved through `WP_Query()` called with `tax_query`?

Reply:
Yeah that's possible too - I took this straight from core WP's `wp_delete_term()` so that we get the identical set of objects that will have the term deleted (where with a WP_Query() you may end up with subtle differences, caching, and pagination).